haystack/nodes/_json_schema.py

```diff
@@ -126,6 +126,39 @@ def find_subclasses_in_modules(importable_modules: List[str]):
     ]
 
 
+def handle_optional_params(param_fields: List[inspect.Parameter], params_schema: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Pydantic v1 cannot generate correct JSON schemas including Optional fields.
+    (https://github.com/samuelcolvin/pydantic/issues/1270)
+    This function detects optional parameters and updates the schema,
+    to allow null values for these parameters.
+    To be removed when Pydantic v2 is released and adopted
+    """
+    optional_params = []
+    for param in param_fields:
+        try:
+            if param.annotation.__origin__ == Union and type(None) in param.annotation.__args__:
+                optional_params.append(param)
+        except:
+            pass
```

Comment:
Why the `try: ... except: pass`? What errors can be raised at this stage? I'd like to have them listed, or at least to see something like:

```suggestion
        try:
            if param.annotation.__origin__ == Union and type(None) in param.annotation.__args__:
                optional_params.append(param)
        except Exception as e: 
            logger.exception("Failed to read parameter type")
```

Reply:
I found a better way to express the conditions, without the need of try/except.